invoice/src/main/java/org/killbill/billing/invoice/tree/AccountItemTree.java

```diff
@@ -57,13 +58,16 @@ public class AccountItemTree {
 
     private boolean isBuilt;
 
-    public AccountItemTree(final UUID accountId, final UUID targetInvoiceId) {
+    private InvoiceConfig invoiceConfig;
+
+    public AccountItemTree(final UUID accountId, final UUID targetInvoiceId, final InvoiceConfig invoiceConfig) {
         this.accountId = accountId;
         this.targetInvoiceId = targetInvoiceId;
         this.subscriptionItemTree = new HashMap<UUID, SubscriptionItemTree>();
         this.isBuilt = false;
         this.allExistingItems = new LinkedList<InvoiceItem>();
         this.pendingItemAdj = new LinkedList<InvoiceItem>();
+        this.invoiceConfig = invoiceConfig;
```

Comment:
Similar remark as for the class `InvoiceDateUtils`. Here we don't offer static method, but these objects are created on the fly with request parameters, and I think it would be best to mix the `InvoiceConfig` - not that most methods would not be available anyways because they need a context, which we don't pass. So here, instead I suggest to pass the integer for `prorationFixedDays`.

Reply:
As discussed, I've updated `AccountItemTree` (and the other related classes) to accept `prorationFixedDays` instead of `invoiceConfig`.